quest_test/test_interrupt_handling.py

```diff
@@ -0,0 +1,95 @@
+import asyncio
+from pathlib import Path
+import shutil
+from src.quest import step, create_filesystem_historian
+import sys
+
+# NOTES:
+# You can run this test without pytest. Error messages will go to err.txt in the working directory
+# When running from pytest, use the "-s" argument to capture output since this test produces many asyncio errors when killing the event loop
+
+sys.stderr = open('err.txt', 'w') # you'll want this file listed in your .gitignore file
+
+# global events to be set by the workflow and indicate how far the workflow progressed
```

Comment:
add in a comment describing the logic of the full testing strategy

Reply:
completed